common/utils/src/main/scala/org/apache/spark/SparkThrowableHelper.scala

```diff
@@ -51,9 +51,11 @@ private[spark] object SparkThrowableHelper {
       messageParameters: Map[String, String],
       context: String): String = {
     val displayMessage = errorReader.getErrorMessage(errorClass, messageParameters)
+    val sqlState = errorReader.getSqlState(errorClass)
+    val displaySqlState = if (sqlState == null) "" else s" SQLSTATE: $sqlState"
     val displayQueryContext = (if (context.isEmpty) "" else "\n") + context
     val prefix = if (errorClass.startsWith("_LEGACY_ERROR_")) "" else s"[$errorClass] "
-    s"$prefix$displayMessage$displayQueryContext"
+    s"$prefix$displayMessage$displaySqlState$displayQueryContext"
```

Comment:
I think should put `displaySqlState` in front of `displayMessage`

Reply:
I disagree. SQLSTATE does not help you read the message. It's 5 random alpha-numeric.

Reply:
@srielau Could you a few sentences to PR's description about reasons for putting this at the end of error messages.

Reply:
@srielau Could you investigate other databases or system? I referenced the Oracle error.